### PR TITLE
add timing terminology

### PR DESCRIPTION
**Pull request by codemaker4**

### CODE REVIEW

**Comment by codemaker4** [comment]
This is based on the discussion we had in Discord and google docs. Those discussions are now discontinued and should continue here.

**Comment by codemaker4** [comment]
PRs should be made based on the discussion it was from, so I am going to mark this as invalid.

**Comment by HerrVincling** [comment]
I don't agree with the definition for ticks.

I think a tick should be defined as the computation and transition to the next logic state. Game ticks are processed at intervals of 25ms. Ticks ≠ Time period.

This also holds up with language:
"After two ticks the signal will arrive here"
"In the next tick this logic gate will turn on"

The benefit is that this matches the definitions you'll find online and the meaning of ticks in the actual programming of the game.

The [Minecraft Wiki](https://minecraft.fandom.com/wiki/Tick) also explains it well.

**Comment by codemaker4** [comment]
Hmm, yeah, it's best to align this with what is used in other places.
Also here is the link to the new wiki. Fandom is discontinued.
https://minecraft.wiki/w/Tick

**Comment by endoddo** [comment]
[In Response to Vincling](https://github.com/Scrap-Mechanic-Logic-Consortium/StandardsDocumentation/pull/3#issuecomment-1955681813)

I first thought that having tick mean the time between calculations would make more sense, since you are typically interested in the state, not the calculation itself.
But as Vincling pointed out, it works well in lanugage and makes more sense when thinking about what the game does.
In most situations and phrases like the ones Vincling mentioned, both would mean the exact same thing regardless of wether a tick would be defined as the moment of the calculation or the period between calculations.

however an expression like:
"In the next tick this logic gate will **be** on"
is confusing when tick = *calculation*, it's unclear whether the gate is on at the start, the end or in the middle of the calculation. 
Whereas the following expression would be confusing for the same reason when a tick is the *state* between calulations.
"In the next tick this logic gate is **turning** on"

Both definitions can be used incorrectly and create confusion. The tick = calculation definition would be easier to describe and is the more correct one. In general I don't think it matters that much since in a lot of cases both work the same and when they don't it's typically clear from context (e.g. present progressive implies the caluclation). So as long as we are consistent it should be fine either way.


*TL;DR: I think the tick = calculation definition is more correct, has the same flaws and benifits in language, just in different situations and oftentimes context helps helps clear up the flaws.*

**Comment by codemaker4** [comment]
> however an expression like:
> "In the next tick this logic gate will be on"
> is confusing when tick = calculation, it's unclear whether the gate is on at the start, the end or in the middle of the calculation.
How about "_After_ the next tick the logic gate will be on"
Or "During the next tick, the logic gate will _turn_ on"
So there are enough ways of saying things that are compatible with the new definition.

**Review by HerrVincling — COMMENTED** [review]
> the instantanious moment that the logic state is updated after the game calculated it

This is still not quite right. The term tick should include the computation.

The Minecraft community has the common abbreviation "MSPT" which stands for **m**illi**s**econds **p**er **t**ick. It is the real time the game takes to compute a tick. In Scrap Mechanic if the MSPT is higher than 25ms you experience lag, same happens in Minecraft because the game simply can't keep up with the tickspeed.

I would phrase it like this:
- A **tick** is defined as the computation of and update to the next logic state. This means a tick is a transition, not a state. Here are some examples: [...]

Overall I don't think any of this really affects casual language, but I do think we should get this right

**Comment by codemaker4** [comment]
Yeah I am bad at wording things lol.
Your wording is now in the standard.
